react-accordion-component/my-app/src/Accordion.tsx

```diff
@@ -0,0 +1,57 @@
+import { MouseEventHandler, useState } from 'react';
+import './Accordion.css';
+
+export type Topic = {
+  title: string;
+  content: string;
+};
+
+type ParentProps = {
+  topics: Topic[];
+};
+
+type ChildProps = {
+  topic: Topic;
+  onToggle: MouseEventHandler;
+  currentItem: boolean;
+};
+
+export default function Accordion({ topics }: ParentProps) {
+  const [openItem, setOpenItem] = useState<null | number>(null);
+
+  function handleToggle(index: number) {
+    if (openItem === index) {
+      return setOpenItem(null);
+    } else {
+      setOpenItem(index);
+    }
+  }
+
+  return (
+    <div className="container">
+      {topics.map((item, index) => (
+        <AccordionItem
+          key={index}
+          onToggle={() => handleToggle(index)}
+          topic={item}
+          currentItem={openItem === index}
+        />
+      ))}
+    </div>
+  );
+}
+
+function AccordionItem({ topic, onToggle, currentItem }: ChildProps) {
```

Comment:
Now that I see how `currentItem` is used, it would probably be better named if it started with `is`, like `isOpen` or `isCurrent`. The name `currentItem` sounds more like it is a a Topic rather than a boolean.